docs/docs/included-server-middleware.md

```diff
@@ -0,0 +1,203 @@
+as to why this header is useful.
+
+```scala mdoc:silent
+import org.http4s.server.middleware.RequestId
+
+val requestIdService = RequestId.httpRoutes(HttpRoutes.of[IO] {
+  case req =>
+    val reqId = req.headers.get(ci"X-Request-ID").fold("null")(_.head.value)
+    // use request id to correlate logs with the request
+    IO(println(s"request received, cid=$reqId")) *> Ok()
+})
+val responseIO = requestIdService.orNotFound(goodRequest)
+
+// generated request id can be correlated with logs
+val resp = responseIO.unsafeRunSync()
+```
+
+Note: `req.attributes.lookup(RequestId.requestIdAttrKey)` can also be used to lookup the request id
+extracted from the header, or the generated request id.
+
+```scala mdoc
+
+// X-Request-ID header added to response
+resp.headers
+// the request id is also available using attributes
+resp.attributes.lookup(RequestId.requestIdAttrKey)
+```
+
+### StaticHeaders
+
+Adds static headers to the response.
+
+```scala mdoc:silent
+import org.http4s.server.middleware.StaticHeaders
+
+val staticHeadersService = StaticHeaders(Headers("X-Hello" -> "hi"))(service).orNotFound
+```
+```scala mdoc
+staticHeadersService(goodRequest).unsafeRunSync().headers
+```
+
+## Request rewriting
+### AutoSlash
+### DefaultHead?
+### HttpMethodOverrider
+### HttpsRedirect?
+### TranslateUri
+### UrlFormLifter
+
+
+## Scaling and management
+### ConcurrentRequests
+### EntityLimiter?
+### MaxActiveRequests
+### Throttle
+### Timeout
+
+## Error handling and Logging
+### ErrorAction
+### ErrorHandling
+### Metrics
+WIP
+
+Apart from the middleware mentioned in the previous section. There is, as well,
+Out of the Box middleware for [Dropwizard](https://http4s.github.io/http4s-dropwizard-metrics/) and [Prometheus](https://http4s.github.io/http4s-prometheus-metrics/) metrics.
+
+
+### RequestLogger, ResponseLogger, Logger
+
+## Advanced
+### BodyCache
+### BracketRequest
+### ChunkAggregator
+### Jsonp
+### ContextMiddleware
+
+## Security
+### CSRF
+
+========================================================
+
+* client mw
+  * CookieJar - allows a client to store and supply cookies
+  * DestinationAttribute
+  * FollowRedirect - allows a client to follow redirects
+  * GZip - allows a client to read gzip/deflate responses
+  * Logger, RequestLogger, ResponseLogger - logs
+  * Metrics - client metrics
+  * Retry - retry requests
+  * UnixSocket - ???
+
+[service]: service.md
+[dsl]: dsl.md
+[Authentication]: auth.md
+[CORS]: cors.md
+[GZip]: gzip.md
+[HSTS]: hsts.md
+[Service Timeout]: @API_URL@/org/http4s/server/middleware/Timeout$
+[Jsonp]: @API_URL@/org/http4s/server/middleware/Jsonp$
+[Virtual Host]: @API_URL@/org/http4s/server/middleware/VirtualHost$
+[Metrics]: @API_URL@/org/http4s/server/middleware/Metrics$
+[`X-Request-ID` header]: @API_URL@/org/http4s/server/middleware/RequestId$
+[Caching]: @API_URL@org/http4s/server/middleware/Caching$.html
```

Comment:
the above links to the javadoc seem to be broken, they eventually redirect to the javadoc for the root package, i think this one is correct